models/ban.js

```diff
@@ -33,11 +35,13 @@ async function nuke(userId, options = {}) {
     const userEvents = await getAllRatingEventsFromUser(userId, options);
 
     for (const userEvent of userEvents) {
-      const eventBalanceOperations = await getRatingBalanceOperationsFromEvent(userEvent.id, options);
+      const promises = [
+        userTabcoin.undoAllByOriginatorId(userEvent.id, options),
+        userTabcash.undoAllByOriginatorId(userEvent.id, options),
+        contentTabcoin.undoAllByOriginatorId(userEvent.id, options),
+      ];
 
-      for (const eventBalanceOperation of eventBalanceOperations) {
-        await balance.undo(eventBalanceOperation.id, options);
-      }
+      await Promise.all(promises);
```

Comment:
Em modo local podemos abrir mais de uma conexão com o PostgreSQL, mas em produção limitamos a uma conexão por lambda, então será preciso executar um por vez.

Mas, na verdade, acho que é melhor não refatorar isso nesse PR.

Reply:
Eu havia mudado do `findAll` com `WHERE originator_id` seguido de um `for` com `balance.undo` para usar direto o `undoAllByOriginatorId` porque senão ficaria assim (em pseudo-código):

```js
const contentBalanceOperations = await getContentBalanceOperationsFromEvent();
const userTabcoinsOperations = await getUserTabcoinsOperationsFromEvent();
const userTabcashOperations = await getUserTabcashOperationsFromEvent();

for (const eventContentBalanceOperations of contentBalanceOperations) {
   await contentTabcoin.undo();
}

for (const eventTabcoinOperations of userTabcoinOperations) {
   await userTabcoin.undo();
}

for (const userTabcashOperations of userTabcashOperations) {
   await userTabcash.undo();
}
```

Ou seja, três `SELECT`s, um para cada tabela, seguido de três laços iterando sobre cada retorno dos `SELECT`s.

Se preferir que o PR 3 (leitura das tabelas novas) fique com o código mencionado acima, e que abra um quarto PR para a refatoração, tudo bem.

Reply:
Por que o PR 3 não poderia permanecer com o código original aqui?

```js
      for (const eventBalanceOperation of eventBalanceOperations) {
        await balance.undo(eventBalanceOperation.id, options);
      }
```

Minha sugestão original (criar novas tabelas no banco) não envolvia a criação de novos *models*. Isso é mesmo necessário?

Reply:
Não é necessário criar novos models se o `models/balance` lidar com o "malabarismo" de usar a tabela adequada dependendo dos argumentos recebidos na função em questão. Seria um model conhecendo a estrutura de três tabelas diferentes, similar à função `rate` criada em `models/content`, porém para todas funções que forem necessárias (`create`, `getTotal`, `undo` e `findOne`, que é usada apenas pelo próprio `undo`).

Pensei que fizesse mais sentido separar a responsabilidade para models diferentes.

Reply:
> Pensei que fizesse mais sentido separar a responsabilidade para models diferentes.

Sim Rafa, deve fazer sentido separar os *models*, mas não necessariamente por estamos separando as tabelas. Apenas acredito ser melhor separar o que não é necessário ao processo de separação das tabelas